src/client/grpc_client.erl

```diff
@@ -761,13 +781,18 @@ call(Process, Request, Timeout) ->
             erlang:demonitor(Mref, [flush]),
             Reply;
         {'DOWN', Mref, _, _, Reason} ->
-            exit(Reason)
+            call_result(ExitOnCrash, Reason)
     after Timeout ->
               erlang:demonitor(Mref, [flush]),
               {error, {grpc_utils:codename(?GRPC_STATUS_DEADLINE_EXCEEDED),
```

Comment:
Unrelated to the changes herein, but there might be a race condition where the timeout is triggered, we demonitor the reference, but a response arrived in the mailbox.

Reply:
✅